lib/background_worker/base.rb

```diff
@@ -1,70 +1,64 @@
 module BackgroundWorker
   class Base
-
     attr_accessor :uid, :state
-    cattr_accessor :logger
-
-    delegate :report_successful, :report_failed, :to => :state
-
 
-    def initialize(options={})
+    def initialize(options = {})
       Time.zone = Setting.time_zone
 
       @uid = options['uid']
 
       # Store state persistently, to enable status checkups & progress reporting
       @state = BackgroundWorker::PersistentState.new(@uid, options.except('uid'))
-      log("Created #{self.class.to_s}")
+      log("Created #{self.class}")
       log("Options are: #{options.pretty_inspect}")
     end
 
-
     # Report progress...
     def report_progress(message)
       state.message = message
       state.save
     end
 
-
     # Report a minor progress -- may get called a lot, so don't save it so often
     def report_minor_progress(message)
       state.message = message
 
       # Only report minor events once per second
       @last_report ||= Time.now - 2
-      if Time.now - @last_report > 1
-        @last_report = Time.now
-        state.save
-      end
-    end
+      time_elapsed =  Time.now - @last_report
+      return unless time_elapsed > 1
 
+      @last_report = Time.now
+      state.save
+    end
 
-    def report_successful(message="Finished successfully")
+    def report_successful(message = 'Finished successfully')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.